ModTek/Logging/BetterLogFormatter.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using System.Diagnostics;
+using Harmony;
+using HBS.Logging;
+using UnityEngine;
+using Logger = HBS.Logging.Logger;
+using Object = UnityEngine.Object;
+
+namespace ModTek.Logging
+{
+    public class BetterLogFormatter
+    {
+        public string LogLineFormat { get; set; } = "{0} [{1}] {2} {3}{4}";
+
+        internal string GetFormattedLogLine(string name, LogLevel logLevel, object message, Object context, Exception exception, IStackTrace location)
+        {
+            var line = string.Format(LogLineFormat,
+                GetFormattedStartupTime(),
+                GetFormattedLogLevel(logLevel),
+                GetFormattedName(name),
+                GetFormattedMessage(message),
+                exception == null ?  GetFormattedLocation(location) : GetFormattedException(exception)
+            );
+
+            return line;
+        }
+
+        public string StartupTimeFormat { get; set; } = "{1:D2}:{2:D2}.{3:D3}";
+
+        private string GetFormattedStartupTime()
+        {
+            var value = TimeSpan.FromSeconds(Time.realtimeSinceStartup);
```

Comment:
using absolute times makes no sense for me, so I use the startup time from unity. that way crash patterns during startup can be more easily detected and the time index numbers are pretty low at the beginning

Reply:
That's a pretty good idea, and having both sometimes might be handy. From a straight utility standpoint time since start is probably most useful.

Reply:
I made the formatter very configurable, so I could just add a absolute time format that just never gets added to the line by default. I didn't because of performance reasons, maybe a flag to just switch between the two.
Maybe Morphyum is one of the modders who could benefit from absolute times, since he uses external tools and servers, and it would be easier to sync the logs then.